### PR TITLE
Community Page

### LINKED ISSUE #110: Community Page
## Summary
Create the community page that displays the available communities and allows searching

## Technical Details
Frontend
You can filter by the name.
Use filter function for arrays
Map works inside of rendering to map array elements to components
Remind me to create mock communities

## Acceptance Criteria
- Communities are displayed on the page
- Search function narrows down communities

## Figma
image

### PR DESCRIPTION
**Pull request by Linkaloo**
## Description
Updates the community page component to have the various communities from the database as well as a search function.

## Related Issue
Closes #110 

## Type of Changes
Please select the relevant option:
- [ ] Bug fix
- [x] New feature
- [x] Enhancement
- [ ] Documentation update
- [ ] Other (please specify)
      
I have updated the community model and the endpoint for retrieving them. The community model now contains a longitude, latitude, and radius defining where a community is. It also has a name for the community. 
I have updated the endpoint to retrieve user associations and just return the first name of those associations.
I have made the page mobile responsive as well.
I also have figured out we are going to have to set a community for a user somehow, someway, I thought of doing it by longitude and latitude but maybe thats not the best, so we can talk about that.

## Things to Update
Please add any features, buttons, or pages that might need updating in the future:
- [ ] Button Functunality 
- [ ] Code Documentation 
- [ ] Mobile Responsiveness 
- [x] Code testing 
- [x] Other (please specify)

The other thing left to do is to hook up clicking one of the cards with moving to a specific communities page.

## Screenshots 
Desktop
image

Mobile
image

## Additional Notes
I created the communities through the database. I will work on a way to more easily add communities. I will also create some communities on the hosted database for us.

### CODE REVIEW

**Review by morenj — APPROVED** [review]
Great work!

**Review by Aligary — APPROVED** [review]
Great work! Seeing how you connect the frontend to the backend is very helpful and will make connecting the rest of the app easy.

**Review by KingJordan152 — APPROVED** [review]
Overall, this looks very good. I left a few comments, but they're not pertinent, especially since we're rapidly approaching the deadline. Approving for now! Thank you for your hard work, Josh.

File: client/src/components/CommunityCard/CommunityCard.tsx [hunk 1]
```diff
@@ -9,39 +9,55 @@ import {
...
               </Typography>
               <CardActions disableSpacing>
                 <AvatarGroup max={3} spacing="small">
-                  <Avatar sx={{ bgcolor: red[500] }}>K</Avatar>
-                  <Avatar sx={{ bgcolor: blue[500] }}>J</Avatar>
-                  <Avatar sx={{ bgcolor: orange[500] }}>P</Avatar>
+                  {first3Users.map((name) => (
+                    <Avatar key={name} sx={{ bgcolor: red[500] }}>
+                      {name[0].toUpperCase()}
+                    </Avatar>
```
[the remark this answers is not in the document]
Comment (KingJordan152) [reply client/src/components/CommunityCard/CommunityCard.tsx:49]: This was supposed to be a comment. My apologies.

File: client/src/pages/Community/Community.tsx [hunk 2]
```diff
@@ -1,10 +1,46 @@
...
 
 export default function Community(): JSX.Element {
+  const [search, setSearch] = useState('');
+  const { data: communities, isLoading } = useGetAllCommunitiesQuery();
+
   return (
     <div className={styles['Community']}>
-      <p>Community Page</p>
+      <h1 className={styles['Header']}>Communities</h1>
+      <TextField header="Search" setInputValue={setSearch} />
```
[the remark this answers is not in the document]
Comment (KingJordan152) [reply client/src/pages/Community/Community.tsx:13]: This was supposed to be a comment. My apologies.

**Pull request merged**